app/policies/synapse_policy.rb

```diff
@@ -7,7 +7,7 @@ def resolve
       return scope.where(permission: visible) unless user
 
       scope.where(permission: visible)
-           .or(scope.where(defer_to_map_id: user.shared_maps.map(&:id)))
+           .or(scope.where(defer_to_map_id: user.all_accessible_maps.map(&:id)))
```

Comment:
maybe something like "map is visible if 1) it's public/commons, 2) it's defer_to_map_id isn't nil and points to a map visible to the user, or 3) the user owns the map"

maybe we should even make that note about not being nil explicit in the code

    .or(scope.where.not(defer_to_map_id: nil).where(defer_to_map_id: user.all_accessible_maps.map(&:id))